test/unit/test_dss_client.py

```diff
@@ -279,6 +279,17 @@ def test_manifest_download_bundle(self, _, __):
         self.dss.download_manifest(self.manifest_file, 'aws')
         self._assert_all_files_downloaded()
         self._assert_manifest_updated_with_paths()
+        self._assert_links()
+
+    @patch('hca.dss.DSSClient.get_bundle', side_effect=_fake_get_bundle)
+    @patch('hca.dss.DSSClient._download_file', side_effect=_fake_download_file)
+    def test_manifest_download_bad_file(self, _, __):
+        """
+        Ensure error is raised if a user created file has the same name as the one
+        we're trying to download.
+        """
+        _touch_file(os.path.join(self.manifest[1][0], self.manifest[1][3]))
+        self.assertRaises(RuntimeError, self.dss.download_manifest, self.manifest_file, 'aws')
```

Comment:
Might be nice to have a bit more evidence that the root exception was indeed the IOError.

What I've done in the past, is to either `raise RuntimeError('Bla', list_of_underlying_exceptions[0:10])` and in the assertion examine the `args` attribute. Note 

```
>>> RuntimeError('bla', [IOError()]).args
('bla', [OSError()])
```

Reply:
@hannes-ucsc I'm not sure I understand what you're suggesting here.